routing/dht/lookup.go

```diff
@@ -40,9 +40,12 @@ func (dht *IpfsDHT) GetClosestPeers(ctx context.Context, key key.Key) (<-chan pe
 		peerset.Add(p)
 	}
 
+	// since the query doesnt actually pass our context down
+	// we have to hack this here. whyrusleeping isnt a huge fan of goprocess
```

Comment:
maybe we should have it in process too? (or go back to always-having-a-context-in-a-process and allow:

```
v := px.Context().Value(k)
```